src/scene/lighting/light-texture-atlas.js

```diff
@@ -19,32 +23,69 @@ class LightTextureAtlas {
         this.shadowMapResolution = 2048;
         this.shadowMap = null;
 
+        this.cookieMapResolution = 2048;
+        this.cookieMap = null;
+        this.cookieRenderTarget = null;
+
         // available slots
         this.slots = [];
 
         this.allocateShadowMap(1);  // placeholder as shader requires it
+        this.allocateCookieMap(1);  // placeholder as shader requires it
         this.allocateUniforms();
     }
 
     destroy() {
+        this.destroyShadowMap();
+        this.destroyCookieMap();
+    }
+
+    destroyShadowMap() {
         if (this.shadowMap) {
             this.shadowMap.destroy();
             this.shadowMap = null;
         }
     }
 
+    destroyCookieMap() {
+        if (this.cookieMap) {
+            this.cookieMap.destroy();
+            this.cookieMap = null;
+        }
+        if (this.cookieRenderTarget) {
+            this.cookieRenderTarget.destroy();
+            this.cookieRenderTarget = null;
+        }
+    }
+
     allocateShadowMap(resolution) {
         if (!this.shadowMap || this.shadowMap.texture.width !== resolution) {
+            this.destroyShadowMap();
             this.shadowMap = ShadowMap.createAtlas(this.device, resolution, SHADOW_PCF3);
 
             // avoid it being destroyed by lights
             this.shadowMap.cached = false;
         }
     }
 
+    allocateCookieMap(resolution) {
+        if (!this.cookieMap || this.cookieMap.width !== resolution) {
+            this.destroyCookieMap();
+            this.cookieMap = CookieRenderer.createTexture(this.device, resolution);
+
+            this.cookieRenderTarget = new RenderTarget({
+                colorBuffer: this.cookieMap,
+                depth: false,
+                flipY: true // ///// ????
```

Comment:
Interesting comment! 😆

Reply:
hah yeah, visible to not miss it when creating PR. Didn't work. Fixed.